allennlp_rc/dataset_readers/bert_squad.py

```diff
@@ -0,0 +1,348 @@
+                            token_is_max_context[len(tokens)] = is_max_context
+                            tokens.append(all_doc_tokens[split_token_index])
+
+                        paragraph_len = doc_span.length
+
+                        start_position = None
+                        end_position = None
+                        span_is_impossible = False
+
+                        # For training, if our document chunk does not contain an annotation
+                        # we throw it out, since there is nothing to predict.
+                        doc_start = doc_span.start
+                        doc_end = doc_span.start + doc_span.length - 1
+                        out_of_span = False
+                        if not (tok_start_position >= doc_start and tok_end_position <= doc_end):
+                            out_of_span = True
+                        if out_of_span:
+                            span_is_impossible = True
+                        else:
+                            # we offset by 2 to account for the [CLS] and [SEP] tokens (before/after the question)
+                            # at the beginning of the sequence. NOTE: we don't add the [CLS] token, as it will get
+                            # added later in the indexer process, therefore start_position will be off by 1.
+                            doc_offset = len(query_tokens) + 2
+                            start_position = tok_start_position - doc_start + doc_offset
+                            end_position = tok_end_position - doc_start + doc_offset
+
+                        if span_is_impossible:
+                            continue
+
+                        passage_offsets = []
+                        token_idx = 0
+
+                        for token in doc_tokens:
+                            passage_offsets.append((token_idx, token_idx + len(token)))
+                            token_idx += len(token)
+
+                        instance = self.text_to_instance(
+                            qas_id=qas_id,
+                            question_text=question_text,
+                            passage_tokens=tokens[paragraph_start_id: paragraph_len],
+                            bert_tokens=tokens,
+                            orig_answer_text=orig_answer_text,
+                            start_position=start_position,
+                            end_position=end_position,
+                            answer_texts=[answer["text"] for answer in qa["answers"]],
+                            passage_offsets=passage_offsets,
+                            passage_text=paragraph["context"]
+                        )
+
+                        yield instance
+
+    @overrides
+    def text_to_instance(self,
+                         qas_id: str,
+                         question_text: str,
+                         bert_tokens: List[Token],
+                         passage_tokens: List[Token],
```

Comment:
`passage_tokens` is unused.

Reply:
Thanks for your careful review, I will fix it.